### PR TITLE
feat:[BT-1543]:Add create versions, fix techdebt in scripts

### PR DESCRIPTION
**Pull request by mbatchelor**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: scripts/ci/check-jira-status.sh [hunk 1]
```diff
@@ -6,21 +6,57 @@
...
+if [ ! -d "$HOME/tmp" ]
+then
+  echo "Creating folder $HOME/tmp"
+  mkdir "$HOME/tmp"
+  CREATED="true"
+fi
+resp=$(curl -s https://raw.githubusercontent.com/harness/harness-core/develop/jira-projects.txt --write-out '%{http_code}' --output "$HOME/tmp/jira-projects.txt")
+if [ "$?" -eq 0 ]
+then
+  if [ "$resp" -ne 200 ]
```
Comment (diptiman-adak) [inline scripts/ci/check-jira-status.sh:19]: These two if loops should be combined together with and

File: scripts/ci/message-check.sh [hunk 2]
```diff
@@ -3,16 +3,44 @@
...
+if [ ! -d "$HOME/tmp" ]
+then
+  echo "Creating folder $HOME/tmp"
+  mkdir "$HOME/tmp"
+  CREATED="true"
+fi
+resp=$(curl -s https://raw.githubusercontent.com/harness/harness-core/develop/jira-projects.txt --write-out '%{http_code}' --output "$HOME/tmp/jira-projects.txt")
+if [ "$?" -eq 0 ]
+then
+  if [ "$resp" -ne 200 ]
```
Comment (diptiman-adak) [inline scripts/ci/message-check.sh:16]: ditto

Comment (diptiman-adak) [reply scripts/ci/message-check.sh:16]: Since we are touching these scripts, it will be great to move the common pieces inside a common script

**Review by diptiman-adak — CHANGES_REQUESTED** [review]

File: scripts/ci/check-jira-status.sh (hunk 1 above)
Comment (mbatchelor) [reply scripts/ci/check-jira-status.sh:19]: They cannot be joined into a single statement. These are deciding whether to exit the script. They are logically separate tests:
if curl exits with a non-zero response then exit
if the http response code isn't 200 then exit

**Review by diptiman-adak — APPROVED** [review]

**Comment by mbatchelor** [comment]
retrigger coverage

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**